crates/lib3h/tests/test_suites/two_basic.rs

```diff
@@ -126,6 +135,31 @@ fn test_send_message(alex: &mut NodeMock, billy: &mut NodeMock) {
     assert_eq!(msg.content, response_content);
 }
 
+/// Test SendDirectMessage and response
+fn test_send_message_fail(alex: &mut NodeMock, _billy: &mut NodeMock) {
+    // Send to self
+    let req_id = alex.send_direct_message(&ALEX_AGENT_ID, "wah".as_bytes().to_vec());
+    let (did_work, srv_msg_list) = alex.process().unwrap();
+    assert!(did_work);
+    assert_eq!(srv_msg_list.len(), 1);
+    println!("response: {:?}", srv_msg_list);
+    let msg_1 = &srv_msg_list[0];
+    one_let!(Lib3hServerProtocol::FailureResult(response) = msg_1 {
+        assert_eq!(response.request_id, req_id);
+    });
+
+    // Send to unknown
+    let req_id = alex.send_direct_message(&CAMILLE_AGENT_ID, "wah".as_bytes().to_vec());
+    let (did_work, srv_msg_list) = alex.process().unwrap();
+    assert!(did_work);
+    assert_eq!(srv_msg_list.len(), 1);
+    println!("response: {:?}", srv_msg_list);
+    let msg_1 = &srv_msg_list[0];
+    one_let!(Lib3hServerProtocol::FailureResult(response) = msg_1 {
+        assert_eq!(response.request_id, req_id);
```

Comment:
It might be worth distinguishing the unknown case somehow (an enum variant?) and verifying it in fact was unknown. This could have failed for some other unrelated reason, as far as the test is concerned.

Reply:
yeah, totally but our Error handling code isn't there yet...